tests/test_check_notebook.py

```diff
@@ -175,3 +175,36 @@ def test_check_notebook_preserve_execution_counts(
         notebook, preserve_execution_counts=preserve_execution_counts
     )
     assert output is is_clean
+
+
+@pytest.mark.parametrize(
+    ("notebook_name", "remove_all_notebook_metadata", "is_clean"),
+    [
+        ("clean_notebook_with_notebook_metadata", True, False),
+        ("clean_notebook_with_notebook_metadata", False, False),
+        ("clean_notebook_without_notebook_metadata", True, True),
+        ("clean_notebook_without_notebook_metadata", False, True),
+        ("clean_notebook", True, False),
+        ("clean_notebook", False, True),
+    ],
+)
+def test_check_notebook_remove_all_notebook_metadata(
+    notebook_name: str,
+    *,
+    remove_all_notebook_metadata: bool,
+    is_clean: bool,
+    request: pytest.FixtureRequest,
+) -> None:
+    """Test nb_clean.clean_notebook when removing all notebooks metadata.
+
+    The test with `("clean_notebook_with_notebook_metadata", False, True)` is False due
+    to `clean_notebook_with_notebook_metadata` is not clean. That is because it contains
+    `language_info.version` captured by --preserve-notebook-metadata.
```

Comment:
This is a useful comment! Suggested tweak to wording:
```suggestion
    The test with `("clean_notebook_with_notebook_metadata", False, True)` is False due
    to `clean_notebook_with_notebook_metadata` containing `language_info.version` detected
    when `preserve_notebook_metadata=False`.
```